### DISCUSSION TITLE
"group" distinct value

### DISCUSSION THREAD

**Discussion by aboveyunhai** [observablehq/plot#2196] [Q&A] [answered]
as far as I understand the current `group transform` or `facet`, it will attempt to render multiple groups with the same scale.
Hence the axis will still show even there is no value for it.
I wonder Is it possible to render data and hide the not existing value such that it will something look like this?
This is a "grouping" in a general sense for arbitrary data but not a group we refer to in observable.

I just abuse the observable lib, and fake the gap and title to something I want it to look like:
https://observablehq.com/d/42188b14fd88e012
or maybe I should just render multiple bars chart with all the x/y/domain to be the same, rather than trying to make everything into one plot

![image](https://github.com/user-attachments/assets/301f6d02-8ae5-4951-9e14-416059f91e6e)
![image](https://github.com/user-attachments/assets/2024ffa4-dae8-40f4-bf7a-9f16c9cca0a9)

**Comment by mythmon** [comment, marked as answer]
Having non-shared axes for faceted charts comes up occasionally. The normal answer here is to make multiple charts that each have their own distinct axes, and then control any axis that needs to be shared. Here's an example of that: https://observablehq.com/d/e047d12f66d50640

```js
htl.html`${d3
  .sort(
    d3.groups(data, (d) => d.group),
    (d) => d[0]
  )
  .map(([groupNum, groupData]) =>
    Plot.plot({
      marginLeft: 100,
      x: { grid: true, domain: xDomain },
      y: { label: `letter (group ${groupNum})` },
      marks: [
        Plot.barX(groupData, { y: "letter", x: "frequency", sort: { y: "-x" } })
      ]
    })
  )}`
```

This is common enough that there is probably room for Plot to have some helper, but in the mean time this is what I'd do.

![image](https://github.com/user-attachments/assets/37bfe808-2747-4628-a0de-34b576b48b58)

**Comment by mbostock** [comment]
We call this “faceting with varying scales”. Please upvote https://github.com/observablehq/plot/issues/8 if you are interested in this feature. 🙏
